### PR TITLE
Added backend functionality and backend documentation

### PR DESCRIPTION
**Pull request by erikpersson0884**

### CODE REVIEW

File: Backend/authRouter.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+export const pathToUsersFile = dataPath + '/users.json';
+const pathToAdminKeysFile = dataPath + '/adminKeys.json';
+const lifeTimeForAdminKeys = 10 * 24 * 60 * 60 * 1000; // 10 days in milliseconds
+
+
+initialize_files([], [pathToUsersFile, pathToAdminKeysFile]);
+
+
+authRouter.post('/login', (req, res) => {
+    console.log(req.body)
```
[the remark this answers is not in the document]
Comment (adiwir) [reply Backend/authRouter.js]: we have removed the console.log

File: Frontend/DAT257_Website/vite.config.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-// https://vitejs.dev/config/
+// Vite configuration with selective proxy for /api/ calls
 export default defineConfig({
   server: {
+    port: 3000,
     proxy: {
-      "/api/v1": "http://localhost:8080/",
+      // Proxy all /api/v2/* calls to the specified address
+      "/api/": {
+        target: "http://localhost:8080/",
```
[the remark this answers is not in the document]
Comment (adiwir) [reply Frontend/DAT257_Website/vite.config.ts]: this has been fixed

File: Backend/peopleRouter.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+peopleRouter.post('/addUser', (req, res) => {
+	let people = fs.readFileSync(pathToUsersFile, 'utf8');
+	people = JSON.parse(people);
+
+	let newUser = {
+		name: req.body.name,
+		password: req.body.password,
+		email: req.body.email,
+		accountType: "user",
+		id: Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15)
```
[the remark this answers is not in the document]
Comment (adiwir) [reply Backend/peopleRouter.js]: id's should not be encrypted 🦦

File: README.md [hunk 4]
```diff
@@ -15,12 +15,40 @@
...
+        "name": "John Doe",
+        "email": "johnthecoolguy@gmail.com",
+        "password": "supersecretPassword1234"
+        }
+
+    * /getUsers - Takes a adminKey and returns a list of all users, if the key is valid
+
+    * /addUser - Takes a json object containing name, password and email and creates a new user with accountType "user" and a randomized userId.
+
+    * /updateUser - Takes a user and a adminKey as a json object and updates it if 2 conditions are met:
```
[the remark this answers is not in the document]
Comment (adiwir) [reply README.md:46]: this is an api call

File: Backend/peopleRouter.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	};
+
+	people.push(newUser);
+	fs.writeFileSync(pathToUsersFile, JSON.stringify(people, null, 2), 'utf8');
+	res.status(200).send("user added successfully!");
+});
+
+peopleRouter.post('/removeUser', (req, res) => {
```
[the remark this answers is not in the document]
Comment (erikpersson0884) [reply Backend/peopleRouter.js:71]: yes, by performing a api call to /people/removeUser

File: README.md [hunk 6]
```diff
@@ -15,12 +15,40 @@
...
+
+    * /getUsers - Takes a adminKey and returns a list of all users, if the key is valid
+
+    * /addUser - Takes a json object containing name, password and email and creates a new user with accountType "user" and a randomized userId.
+
+    * /updateUser - Takes a user and a adminKey as a json object and updates it if 2 conditions are met: 
+        - The adminKey is valid
+        - The adminKey matches the userId
+
+    * /removeUser - Takes a adminKey and a userId and removes the user if:
```
[the remark this answers is not in the document]
Comment (adiwir) [reply README.md:50]: this is alsa api calls

File: Backend/peopleRouter.js
[the remark this answers is not in the document]
Comment (erikpersson0884) [reply Backend/peopleRouter.js:1]: :(

**Review by Pontus-Wikstrom — APPROVED** [review]
LGTM

**Pull request merged**